### ISSUE TITLE
[REQUEST] Muon Optimizer - Different LR for Different Groups

### ISSUE THREAD

**Issue by DongmingShenDS** [microsoft/DeepSpeed#7657]
**Is your feature request related to a problem? Please describe.**
When using Muon optimizer, now the method is to setup that in config. Params will be grouped into muon & adam automatically per #7555. However, there's a single LR that passed into both groups - this is not the standard pattern. We should have different LR for the muon group and the adam group (usually, muon should have larger LR). I wonder if there's a way to enable this behavior easier from my side. If not, then we probably need to add a support for different LR. 

**Describe the solution you'd like**
Easiest way to achieve this is probably to have "muon_lr" and "adam_lr" instead of a single "lr" key. Then in different groups, fetch accordingly.

**Comment by delock** [comment]
The problem statement is when optimizer is marked as "muon", actually both adam and muon optimizer will be used -- muon for >2D weights for hidden layers, adam for the rest.  By default "lr" will be learning rate for both optimizer, and this will not be sufficient for advanced users.

The solution could be use "lr" as default learning rate.  And "muon_lr" and "adam_lr" as override.   If one of them is specified, the specified value will be used instead of default learning rate, this makes these two config item optional.

@PKUWZP @sfc-gh-truwase FYI.

**Comment by sowndappan5** [comment]
I agree. Since Muon internally splits parameters into Muon and Adam groups, exposing separate muon_lr and adam_lr configuration options would better align with common Muon training recipes and remove the need for custom optimizer setup. It would also make DeepSpeed's Muon integration more flexible while keeping the user experience simple.

**Comment by sfc-gh-truwase** [comment]
@DongmingShenDS and @sowndappan5 thanks for the proposal of `adam_lr` and `moun_lr` overrides. Are you able to submit a PR? Thanks!

**Comment by sowndappan5** [comment]
Thanks! Yes, I can take a look and work on a PR for this enhancement.

I'll first investigate how the Muon and Adam parameter groups are created and how learning rates are currently propagated through the configuration and optimizer setup. If the change is straightforward and backward-compatible, I'll prepare a PR with support for separate `muon_lr` and `adam_lr` overrides.

I'll post my findings here as I progress.

**Comment by sowndappan5** [comment]
Thanks, I’ve opened a PR for this here: [link.](https://github.com/deepspeedai/DeepSpeed/pull/8047) While tracing the implementation, I found that the core muon_lr/adam_lr override logic is already present in the Muon optimizer path, so this PR adds focused test coverage for that behavior and fixes the related Muon docs wording. The added test verifies backward compatibility with lr as the default for both groups and verifies that muon_lr and adam_lr override the Muon and Adam parameter groups independently.
